source/server/backtrace.h

```diff
@@ -100,23 +117,32 @@ class BackwardsTrace : Logger::Loggable<Logger::Id::backtrace> {
     }
 
     const auto thread_id = stack_trace_.thread_id();
-    // Trick to figure out our own object file name for script to use:
-    backward::ResolvedTrace trace = resolver.resolve(stack_trace_[0]);
-    auto obj_name = trace.object_filename;
-    if (obj_name.empty()) {
-      obj_name = "path/to/envoy-executable";
-    }
+    backward::ResolvedTrace first_frame_trace = resolver.resolve(stack_trace_[0]);
+    auto obj_name = first_frame_trace.object_filename;
 
     LogAtLevel("Backtrace obj<{}> thr<{}> (use tools/stack_decode.py):", obj_name, thread_id);
 
     // Why start at 2? To hide the function call to backward that began the
     // trace.
     for (unsigned int i = 0; i < stack_trace_.size(); ++i) {
+      // Backtrace gets tagged by ASAN when we try the object name resolution for the last
+      // frame on stack, so skip the last one. It has no useful info anyway.
+      if (i + 1 != stack_trace_.size()) {
```

Comment:
I'm a little worried that this is arch specific, and Backwards won't always return this sentinel stack frame as the last item.  But, it should always be _safe_ to skip the last item, the risk is that we don't include some relevant stack frame.  I'll switch to stopping at size-1 which should guarantee no ASAN issues like I saw, at the possible risk of losing the bottom of stack which we probably don't care about anyway.